aten/src/ATen/native/xnnpack/OpContext.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#ifdef USE_XNNPACK
+
+#include <ATen/core/ivalue.h>
+#include <ATen/native/xnnpack/Common.h>
+#include <ATen/Tensor.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+
+using SerializationTypeLinearPrePack = std::tuple<Tensor, c10::optional<Tensor>>;
+using SerializationTypeConv2dPrePack =
+  std::tuple<Tensor, c10::optional<Tensor>,
+  std::vector<int64_t>, std::vector<int64_t>, std::vector<int64_t>, int64_t>;
+
+class XNNPackLinearOpContext : public torch::jit::CustomClassHolder {
+  private:
+    Tensor orig_weight_;
+    c10::optional<Tensor> orig_bias_;
+    Context op_context_;
```

Comment:
OK, so if I am reading the code correctly, it seems we need separate contexts for Linear and Convolution anyway.  I think it would be better just to separate them, and expose them in Convolution.h and Linear.h as opposed to bundling them together in Common.h.  Or maybe I am missing something?

Reply:
I explained the reasoning above, but you are right, we are storing context separately anyway.